src/app/api/widget-execution/route.tsx

```diff
@@ -0,0 +1,133 @@
+import { imageResponseOptions } from 'src/components/ImageGeneration/imageResponseOptions';
+import WidgetExecutionSSR from 'src/components/ImageGeneration/WidgetExecutionSSR';
+
+const WIDGET_IMAGE_WIDTH = 416;
+const WIDGET_IMAGE_HEIGHT = 432;
+const WIDGET_IMAGE_SCALING_FACTOR = 2;
+
+export async function GET(request: Request) {
+  const { searchParams } = new URL(request.url);
+  const amount = searchParams.get('amount');
+  const fromToken = searchParams.get('fromToken');
+  const fromChainId = searchParams.get('fromChainId');
+  const toToken = searchParams.get('toToken');
+  const toChainId = searchParams.get('toChainId');
+  const highlighted = searchParams.get('highlighted');
+  const theme = searchParams.get('theme');
+  const isSwap = searchParams.get('isSwap');
+
+  // Fetch chain data asynchronously before rendering
+  const getChainData = async (chainId: ChainId) => {
+    const chainsData = await getChains({
+      chainTypes: [ChainType.EVM, ChainType.SVM],
+    });
+    return chainsData.find((chainEl) => chainEl.id === chainId);
+  };
+
+  // Fetch from and to chain details (await this before rendering)
+  const fromChain = fromChainId
+    ? await getChainData(parseInt(fromChainId) as ChainId)
+    : null;
+  const toChain = toChainId
+    ? await getChainData(parseInt(toChainId) as ChainId)
+    : null;
+
+  // Fetch token asynchronously
+  const fetchToken = async (chainId: ChainId | null, token: string | null) => {
+    if (!chainId || !token) {
+      return null;
+    }
+    try {
+      const fetchedToken = await getToken(chainId, token);
+      return fetchedToken;
+    } catch (error) {
+      console.error('Error fetching token:', error);
+      return null;
+    }
+  };
+
+  const fromTokenData =
+    fromChainId && fromToken
+      ? await fetchToken(parseInt(fromChainId) as ChainId, fromToken)
+      : null;
+  const toTokenData =
+    toChainId && toToken
+      ? await fetchToken(parseInt(toChainId) as ChainId, toToken)
+      : null;
+
+  const options = await imageResponseOptions({
+    width: WIDGET_IMAGE_WIDTH,
+    height: WIDGET_IMAGE_HEIGHT,
+    scalingFactor: WIDGET_IMAGE_SCALING_FACTOR,
+  });
+
+  return new ImageResponse(
+    (
+      <div
+        style={{
+          position: 'relative',
+          display: 'flex',
+          width: WIDGET_IMAGE_WIDTH * WIDGET_IMAGE_SCALING_FACTOR,
+          height: WIDGET_IMAGE_HEIGHT * WIDGET_IMAGE_SCALING_FACTOR,
+        }}
+      >
+        <img
+          alt="Widget Example"
+          width={'100%'}
+          height={'100%'}
+          style={{
+            margin: 'auto',
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            objectFit: 'cover',
+            width: WIDGET_IMAGE_WIDTH * WIDGET_IMAGE_SCALING_FACTOR,
+            height: WIDGET_IMAGE_HEIGHT * WIDGET_IMAGE_SCALING_FACTOR,
+          }}
+          src={`${process.env.NEXT_PUBLIC_SITE_URL}/widget/widget-execution-${theme === 'dark' ? 'dark' : 'light'}.png`}
```

Comment:
@oktapodia Do you have a good idea on how to make the "baseUrl" dynamic? Although, this woud likely work on environments such as develop, staging or production -> It does not work on localhost (as it´s not picking up  `NEXT_PUBLIC_SITE_URL` from **.env.localhost**. Also, it would be awesome if this could already be tested on a feature-branch, but I have no idea to grab that baseUrl which is interchangeable.

I also tried removing the baseUrl like so `/widget/widget-execution-${theme === 'dark' ? 'dark' : 'light'}.png` but that results in an error since this is not being picked up by the next-js api routes.

THANKS ALOT!!